### DISCUSSION TITLE
v2.8.4

### DISCUSSION THREAD

**Discussion by MAKOMO** [artisan-roaster-scope/artisan#1204] [Versions]
This version comes introduces dark mode on Windows and Linux and adds official support for [Kaleido](https://www.kaleido-roaster.com/) roasting machines.

---
Release Sponsor: [BC Roasters](https://bcroasters.com/)  
<a target="_blank" href="https://bcroasters.com/">
<img src="https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/screenshots/bc-logo.jpg?raw=true" width="500"></a>
---

The [Artisan project](https://artisan-scope.org/) runs on donations from individuals and companies recognising the value of Artisan for their work or leisure. Consider supporting this project with your [donation](https://www.paypal.com/paypalme/MarkoLuther), or even better, purchase an [artisan.plus membership](https://shop.artisan.plus/)!

## Documentation

__>>>  [Release Notes](https://artisan-roasterscope.blogspot.com/2023/06/artisan-v284.html) <<<__

See the [detailed release history](https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/ReleaseHistory.md) for a complete list of additions, changes and fixes and the [Installation Instructions](https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/Installation.md) for platform specific notes. Read our [Quick-Start Guide](https://artisan-scope.org/docs/quick-start-guide/)!

## Supported Platforms

Builds are available for macOS 11 Big Sur, Windows 10 (x64), Ubuntu 22.04+ (GCC11), Red Hat 8.4+ (GCC10). An RPi 64bit build will be added once Raspberry Pi OS 12 bookworm is released.

Legacy builds for macOS High Sierra 10.13, Windows 8 (x64) are available too.

---

PLEASE FOLLOW AND TAG US!  
&nbsp;&nbsp;&nbsp;&nbsp;<a href="https://www.facebook.com/ArtisanScope"><img src="https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/screenshots/facebook-square.svg?raw=true" width="30"></a>&nbsp;&nbsp;&nbsp;<a href="https://www.instagram.com/artisanscope/"><img src="https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/screenshots/instagram-square.svg?raw=true" width="30"></a>&nbsp;&nbsp;&nbsp;<a href="https://www.youtube.com/channel/UCEp-DH_Y7sBR3-X3FWomgog"><img src="https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/screenshots/youtube-svgrepo-com.svg?raw=true" width="30"></a>&nbsp;&nbsp;&nbsp;<a href="https://masto.ai/@artisan"><img src="https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/screenshots/mastodon-icon-svgrepo-com.svg?raw=true" width="30"></a>&nbsp;&nbsp;&nbsp;<a href="https://www.linkedin.com/company/artisan-plus"><img src="https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/screenshots/linkedin-svgrepo-com.svg?raw=true" width="30"></a>
---

__Licence Notes__

_Artisan is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version._

_Artisan is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE._

_Copies of the GNU General Public License has been included with this 
distribution in the file `LICENSE.txt`. An online version is available at
<http://www.gnu.org/licenses/>._

<hr /><em>This discussion was created from the release <a href='https://github.com/artisan-roaster-scope/artisan/releases/tag/v2.8.4'>v2.8.4</a>.</em>

**Comment by aday2023** [comment]
Bonsoir tout le monde,
Je suis là pour deux raison et j'espère avoir une solution:
1- torréfaction de marque HAS GARANTI arrive pas a se connecter avec ARTISAN a chaque fois le message : modbus communication error.
2- Je sais pas le model de ma machine car je l'ai acheté d'occasion.
Merci pour vos réponses.

**Reply by poundy** [reply]
hi. 
You should start your own topic. This is the 2.8.4 release and your issues have nothing to do with that.
Has Garanti are a supported machine for some configurations. https://artisan-scope.org/machines/hasgaranti/. If you don't know your hardware, we won't know how to connect it! Modbus is simple. If you have a USB connection to the roaster, then you would be able to figure it out. Lets help in your new topic.

**Comment by boblowkl** [comment]
Umm... Raspberry Pi Bookworm was launched a week ago. 😬 pretty please with sugar on top?

**Reply by MAKOMO** [reply]
There is a test build of the next Artisan version which should run on Bookworm (64bit).

https://github.com/artisan-roaster-scope/artisan/releases/tag/RPi-test

Please report!

**Reply by shawnaks5** [reply]
Hello.  Does anyone know how to make Artisan work on Mac OS 14 Sonoma?  I just upgraded my computer and now I'm getting the modbus fail issue.  I am new to GitHub, am not a computer expert at all and know nothing about settings or anything like that.  My email: Shawn.c.adkins@gmail.com.   Need help please!

**Reply by MAKOMO** [reply]
Hacking a thread this way is never ok. There is a test build of the next Artisan which works stable on Sonoma. Thus your MODBUS communication issues might not be resolved by this build and are rather a configuration issue.

https://github.com/artisan-roaster-scope/artisan/releases/tag/m1-test5

Please use open a separate thread to ask the community for support on your "modbus fail issue" what ever that might be, and don't forget to supply all the relevant information.

PS: Never publish your email online if you are slow in reading SPAM!

**Reply by shawnaks5** [reply]
I appreciate the response!  I apologize!  I wasn't aware that I was hacking a thread.   Not my intention at all.  I am new to GitHub and frankly don't know too much about computers. 

I tried that version of Artisan you are speaking of and that didn't work.  Not sure how to change the configuration of my computer to make that version work and I'm not sure what "relevant" info I need to post in my question in order to get help.  

Again, I apologize for putting my question in the wrong place.  Just trying to get things back online as quickly as possible.

**Comment by KeithInvan** [comment]
Could we please have the .deb and AppImage files in Assets?  Only the source code is there now.  Binaries for 2.10.2 are not working for me in Debian nor Linux Mint.

**Reply by MAKOMO** [reply]
Outdated Artisan versions are unsupported. If you have an issue with the current version, please file a corresponding issue and we will fix it. Thanks!

**Reply by KeithInvan** [reply]
Right.  Completely understand.  Not looking for support.  Was wishing to do some backtesting to determine which versions were not happy with my setups (few different Linux platforms, no Windows, no Mac), then open a report to you for further discovery or consideration.  Latest version is not happy for sure, both the AppImage and the dpkg.

**Reply by MAKOMO** [reply]
Could you PLEASE open an issue on this and report the terminal output on starting Artisan 2.10.2 without the startup script like
```
# /usr/share/artisan/artisan
```
and document which OS you are running and which WindowServer (Wayland or X11) . Thanks for your support!

**Reply by MAKOMO** [reply]
Maybe you can also try out our current continuous build which might have fixed the issue already: https://github.com/artisan-roaster-scope/artisan/releases

**Reply by KeithInvan** [reply]
To confirm, 2.10.3 works flawlessly for me on Linux Mint 21.3 , Kernel: Linux 6.5.0-21-generic, Architecture: x86-64. xorg-server 2:21.1.4-2ubuntu1.7~22.04.8.  Thank you for pointing out the continuous build was available.

**Comment by Xuxiaobu11** [comment]
How  do  I  download  version  2.8.4?

**Reply by MAKOMO** [reply]
What for? Just download the [latest Artisan](https://github.com/artisan-roaster-scope/artisan/releases/latest)!

**Reply by Xuxiaobu11** [reply]
Ok,thank  you

**Reply by alyasik1** [reply]
My roaster only supports version 2.8.4 how to download

**Reply by MAKOMO** [reply]
Who is that? Current Artisan versions support everything which v2.8.4 supported. What roasting machine do you have which is not supported by the current release?

**Reply by alyasik1** [reply]
Sorry, everything is ok, misunderstood the documentation

**Reply by alyasik1** [reply]
Probat P-Series P12 III Still can't install, maybe you have some more extensive instructions?

**Reply by poundy** [reply]
> Probat P-Series P12 III Still can't install, maybe you have some more extensive instructions?

Create a new discussion and document what you are doing and what you have tested and what "still can't install" actually means ! This thread is about the release of a now outdated, no longer supported release, and is not intended for disussing installing new versions and troubleshooting that kind of issue.

**Comment by mehrdadpirnazar** [comment]
Hello Sliders do not work properly on touch screens Please fix it

@MAKOMO

**Reply by MAKOMO** [reply]
@mehrdadpirnazar : a bit pushy. Why not fix it yourself. This is after all an open-source project where everybody is invited to contribute! You even have the hardware which I don't.

**Reply by mehrdadpirnazar** [reply]
I apologize if my speech was offensive But unfortunately, we have been facing this problem for two months and several experts were not able to solve this problem

**Reply by mehrdadpirnazar** [reply]
I can help you in this way if you do not have access to a touch screen If you have time and you think the sliders will improve with some changes I will work with you to find the problem

**Reply by poundy** [reply]
for a start, lets not pollute other un-related discussions with your "problem". Stay in the one place you should, your own thread... https://github.com/artisan-roaster-scope/artisan/discussions/1549#discussioncomment-9424897

**Comment by thedirtyduo** [comment]
Hi I'm on OSX 11.7.10 and have updated to 2.10.4 (legacy). I'm not able to go through the standard steps to get the roaster heating up using a saved profile. I make sure it says "Super On" and then when I hit "Control" I get nothing. Can I also looked for a direct link to the 2.8 version .dmg file but can't find it. Anyone have a link to that older version?

**Reply by MAKOMO** [reply]
@thedirtyduo : Could you please add an issue giving all the details that might be necessary to work this out? Please include your Artisan settings (uploaded via drag-and-drop after exporting, menu Help >> Export Settings, and renaming to artisan-settings.txt), that saved profile you are using for roaster heating up as .alog file (again drag-and-dropped and renamed to end in *.txt), a copy of your message log (menu Help >> Messages) for a start. More information like log files might be requested later.

Older Artisan versions are not available for download. Sorry. Did Artisan v2.8.? worked for you before?
